src/Package/Impl/DataInspect/DataSource/GridDataSource.cs

```diff
@@ -13,14 +14,8 @@ namespace Microsoft.VisualStudio.R.Package.DataInspect.DataSource {
     public class GridDataSource {
         public static async Task<IGridData<string>> GetGridDataAsync(string expression, GridRange? gridRange, IRSession rSession = null) {
             await TaskUtilities.SwitchToBackgroundThread();
-
-            if (rSession == null) {
-                rSession = VsAppShell.Current.ExportProvider.GetExportedValue<IRSessionProvider>().GetInteractiveWindowRSession();
-                if (rSession == null) {
-                    throw new InvalidOperationException(Invariant($"{nameof(IRSessionProvider)} failed to return RSession for {nameof(VariableViewModel)}"));
-                }
-            }
-
+            rSession = rSession ?? VsAppShell.Current.ExportProvider.GetExportedValue<IRSessionProvider>().GetInteractiveWindowRSession();
```

Comment:
It is passed as null in data import dialog. Not entirely sure why, we can probably change this and have it always non-null